### ISSUE TITLE
FEATURE: Detect first login

### ISSUE THREAD

**Issue by Aradon267** [magshimim-next/cv-next#201]
Find a way to see if it's the user's first login(this means that it's after the user was approved by the mods i.e. it's an active user's first login).

After that we can prompt it o change it's username and in the future add a feature guide or something

**Comment by lordYorden** [comment]
updated_at field should be just created at, it doesn't update with the table in the firebase auth

**Comment by lordYorden** [comment]
as for checking the first login we can use this field to compare it with `now()`

**Comment by lordYorden** [comment]
used `superbase.auth.getUser()` and and created component that check `user.last_sign_in_at == user.created_at` on client side
![WhatsApp Image 2024-10-31 at 17 53 35_5270dbb0](https://github.com/user-attachments/assets/e3bc2620-b634-47a8-86ea-677ae94efca9)

**Comment by lordYorden** [comment]
@Aradon267  I fixed some issues with the tooltip component (added hover and children)
but it caused some weird behavior with comments 
![WhatsApp Image 2024-10-31 at 17 54 10_b61b2ac2](https://github.com/user-attachments/assets/1115b540-51cf-4019-a595-36955cebb8f9)
tried to fix it with no luck

**Comment by lordYorden** [comment]
fixed
![image](https://github.com/user-attachments/assets/f2360b8c-7034-412e-8c29-d927692bc659)

**Comment by lordYorden** [comment]
and this how the sign in tooltip looks
![image](https://github.com/user-attachments/assets/439ba88a-b4e1-4e44-a8a1-70f0b67fd067)

**Comment by Aradon267** [comment]
Won't the created at be affected even if the user wasn't active? 
Other than that, looks good!

**Comment by lordYorden** [comment]
> Won't the created at be affected even if the user wasn't active? Other than that, looks good!

don't know we need to find a viable way to test it, according to documentation it won't

**Comment by Aradon267** [comment]
I'll give it a go with a different Gmail I own

**Comment by lordYorden** [comment]
just change the `line 16` on `firstTimeSignIntsx`

**Comment by Aradon267** [comment]
I logged in and out several times and I don't see the thing there

**Comment by lordYorden** [comment]
you wont see it there except the very first time you log in.
that is way it is very hard to test. we can just detect the hash and prompt it then. 
so it will appear until you update your username.

**Comment by lordYorden** [comment]
could be a better alternative

**Comment by lordYorden** [comment]
```ts
const user = await getUser();
if (!user.ok) return; //TODO: maybe push to inactive page
const username = user.val.username;
setIsFirstLogin(username !== undefined && username !== null && username.includes("."));
```

is something like this is better?

**Comment by Aradon267** [comment]
Basing it on the dots? It might be a bit risky as I'm not sure it's problematic(people might actually use dots in their names) @YechiamW3 what do you think?

**Comment by lordYorden** [comment]
@Aradon267 suggested we add a field to superbase's user metadata for `isFirstLogin`

**Comment by Aradon267** [comment]
This is what I meant
https://supabase.com/docs/reference/javascript/auth-updateuser

**Comment by YechiamW3** [comment]
1. @lordYorden regarding checking against generated username - I think the only real way to do it is to have another field that will always have that original generated username, and compare the current one against it. Any other check cannot be trusted - we don't know what username the user might give.
2. Regarding `isFirstLogin` field - I want to make sure I understand @Aradon267's suggestion: on the auth callback route we check if `isFirstLogin` is true, then e.g. make the redirect with another query param of `first-login`, catch it in the app and continue from there (e.g. "change username" popup that redirect to profile page with edit=true and username validation or smth like that)?
Yeah it sounds good; we just need to make sure it can't be abused easily - e.g. if we truly use the query-param method, we need to make sure it's truly `firstLogin=true` and only change it to false afterwards.

In summary, the flow I'll suggest will be:
1. auth callback -> if `firstLogin=true`: redirect with query param `firstLogin=true`
2. in landing page, catch the `firstLogin` query param, if exists (&==true), do one of the following:
  2.1. produce a popup with username change requirement
  2.2. produce a popup that states a username change is required with a button that redirects to profile page with edit=true and a username validation (required & not like previous, uniquity validation will be done in server-side)
3. after successful username change, change `firstLogin` property to `false`

Sounds good? Any counter-suggestions/other ideas?
-

**Comment by lordYorden** [comment]
@YechiamW3  sounds great, but we can use the new `full_name` we discussed on #202 
to check if it's the same as the username and It makes **option 1** the most viable option 
that seems the easiest and simplest way to implement this

**Comment by YechiamW3** [comment]
> @YechiamW3 sounds great, but we can use the new `full_name` we discussed on #202 [this thread](https://github.com/magshimim-next/cv-next/issues/202#issuecomment-2453467140) to check if it's the same as the username and
> 
> ### It makes option 1 the most viable option that seems the easiest and simplest way to implement this

Wait, what would comparing `full_name` and `username` give you?
`full_name` has not relation whatsoever to `username` - and it shouldn't have, `username` is unique, while `full_name` is not.

**Comment by lordYorden** [comment]
I ment that we take the full name and use it as the display name as discussed in #202. At the beginning we set it to the generated username
And check if it is still the same.

If you want to go with option 2 it is fine by me. Just wanted you to understand what I ment.

**Comment by YechiamW3** [comment]
Ah, I don't think putting the display name with that ugly generated value is nice for the user.
In any case, I think opt.2 is more preferable either way.

**Comment by lordYorden** [comment]
@YechiamW3 I agree the question now is what we define as a `firstLogin`. is it until the `username/display_name` (depending on 
#218) changes or only on first time the user logs in.

and if it is **option 1**, should it be more obvious? like `wasGenerated` or `isChanged`

**Comment by Aradon267** [comment]
If we do go with 2 it should be based on the "true" login state(only after the user was activated and logged in)

**Comment by YechiamW3** [comment]
I agree that the second option is better. It also eliminates the need for problematic checks for a generated username.

**Comment by lordYorden** [comment]
![image](https://github.com/user-attachments/assets/790d78bb-101e-4052-96bd-62b0e90e4072)

new username screen
